experimental/dds/tree2/src/core/tree/deltaUtil.ts

```diff
@@ -3,10 +3,67 @@
  * Licensed under the MIT License.
  */
 
-import { Mark, Root, Skip } from "./delta";
+import { Mutable } from "../../util";
+import { ITreeCursorSynchronous } from "./cursor";
+import { Root, DetachedNodeId, FieldChanges } from "./delta";
+import { rootFieldKey } from "./types";
 
 export const emptyDelta: Root<never> = new Map();
 
-export function isSkipMark(mark: Mark<unknown>): mark is Skip {
-	return typeof mark === "number";
+export const emptyFieldChanges: FieldChanges<never> = {};
+
+export function isEmptyFieldChanges(fieldChanges: FieldChanges): boolean {
+	return (
+		fieldChanges.attached === undefined &&
+		fieldChanges.detached === undefined &&
+		fieldChanges.build === undefined &&
+		fieldChanges.rename === undefined
+	);
+}
+
+export function deltaForRootInitialization(content: readonly ITreeCursorSynchronous[]): Root {
+	if (content.length === 0) {
+		return emptyDelta;
+	}
+	const buildId = { minor: 0 };
+	const delta: Root = new Map([
+		[
+			rootFieldKey,
+			{
+				build: [{ id: buildId, trees: content }],
+				attached: [{ count: content.length, attach: buildId }],
+			},
+		],
+	]);
+	return delta;
+}
+
+export function deltaForSet(
+	newNode: ITreeCursorSynchronous,
+	buildId: DetachedNodeId,
+	detachId?: DetachedNodeId,
+): FieldChanges {
+	const mark = { count: 1, attach: buildId, detach: detachId };
+	if (detachId === undefined) {
+		delete mark.detach;
+	}
```

Comment:
```suggestion
	const mark: Mark = { count: 1, attach: buildId };
	if (detachId !== undefined) {
		mark.detach = detachId;
	}
```

Seems more natural to me.